### PR TITLE
Rimworld Pack Rework

### PR DESCRIPTION
**Pull request by Laugic**

### CODE REVIEW

File: src/main/java/thePackmaster/cardmodifiers/rimworldpack/OnFireModifier.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        return true;
+    }
+
+    @Override
+    public AbstractCardModifier makeCopy() {
+        return new OnFireModifier();
+    }
+
+}
```
Comment (Laugic) [inline src/main/java/thePackmaster/cardmodifiers/rimworldpack/OnFireModifier.java:109]: This was copied and trimmed from the existing code for Infest counters

File: src/main/java/thePackmaster/actions/rimworldpack/KillThirstAction.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.vfx.combat.FlashAtkImgEffect;
+import thePackmaster.cards.rimworldpack.Despair;
+
+public class KillThirstAction extends AbstractGameAction {
+
+    private DamageInfo info;
+
+    //Copied from Sunder but changed to lose Mood if not fatal instead of gain E if fatal
+    public KillThirstAction(AbstractCreature target, DamageInfo info, int moodAmount) {
```
Comment (erasels) [inline src/main/java/thePackmaster/actions/rimworldpack/KillThirstAction.java:17]: Mood amount is probably not needed here

File: src/main/java/thePackmaster/cards/rimworldpack/EatWithoutTable.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        //Drink a random potion
+        addToBot(new AbstractGameAction() {
+            @Override
+            public void update() {
+                AbstractPotion potion = null;
+                int tries = 0;
+                while(tries < 100 && potion == null){
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rimworldpack/EatWithoutTable.java:38]: This is pretty jank. Please just create a list of potions by iterating through PotionHelp.potions. Then take a random one from that and make a copy of it

File: src/main/java/thePackmaster/patches/rimworldpack/FreeToPlayPatch.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import com.megacrit.cardcrawl.ui.panels.EnergyPanel;
+import thePackmaster.cards.rimworldpack.FreeToPlayPower;
+import thePackmaster.util.Wiz;
+
+public class FreeToPlayPatch {
+    @SpirePatch2(clz = AbstractCard.class, method = "freeToPlay")
+    public static class FreeCardPlz {
+        @SpirePrefixPatch
+        public static SpireReturn<?> free(AbstractCard __instance) {
+            if (Wiz.adp() != null && !AbstractDungeon.isScreenUp && AbstractDungeon.currMapNode != null && AbstractDungeon.getCurrRoom().phase == AbstractRoom.RoomPhase.COMBAT) {
```
Comment (erasels) [inline src/main/java/thePackmaster/patches/rimworldpack/FreeToPlayPatch.java:20]:
```suggestion
            if (Wiz.isInCombat() && !AbstractDungeon.isScreenUp) {
```
requires import

File: src/main/resources/anniv5Resources/localization/eng/rimworldpack/Cardstrings.json [hunk 5]
```diff
@@ -1,32 +1,48 @@
 {
   "${ModID}:ThrowHorseshoe": {
     "NAME": "Throw Horseshoe",
-    "DESCRIPTION": "Deal !D! damage. NL Discard a random card. NL ${ModID}:Increase_Mood by !M!."
+    "DESCRIPTION": "Deal !D! damage. NL ${ModID}:Increase_Mood by !M!."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rimworldpack/Cardstrings.json:4]: This card doesn't exist anymore

File: src/main/resources/anniv5Resources/localization/eng/rimworldpack/Cardstrings.json [hunk 6]
```diff
@@ -1,32 +1,48 @@
...
     "DESCRIPTION": "Deal !D! damage to a random enemy !M! times."
   },
+  "${ModID}:Despair": {
+    "NAME": "Despair",
+    "DESCRIPTION": "Counts as a Curse. NL Exhaust."
+  },
   "${ModID}:Relax": {
     "NAME": "Relax",
-    "DESCRIPTION": "{@@}Discard !${ModID}:m2! cards. NL ${ModID}:Increase_Mood by !M!."
+    "DESCRIPTION": "Discard any number of cards. ${ModID}:Increase_Mood by !M!."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rimworldpack/Cardstrings.json:16]: Please remove the cards that don't exist anymore

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/cards/rimworldpack/EatWithoutTable.java [hunk 7]
```diff
@@ -34,14 +36,17 @@ public void use(AbstractPlayer p, AbstractMonster m) {
...
-                     AbstractPotion newTry = PotionHelper.getRandomPotion();
-                     if(newTry.rarity == AbstractPotion.PotionRarity.COMMON && newTry.canUse() && !newTry.isThrown)
-                         potion = newTry;
-                     else
-                         tries++;
+                ArrayList<AbstractPotion> potions = PotionHelper.getPotionsByRarity(AbstractPotion.PotionRarity.COMMON);
+                ArrayList<AbstractPotion> drinkable = new ArrayList<>();
+
+                for (AbstractPotion pot: potions) {
+                    if(pot.rarity == AbstractPotion.PotionRarity.COMMON && pot.canUse() && !pot.isThrown)
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rimworldpack/EatWithoutTable.java:43]:
```suggestion
                    if(pot.canUse() && !pot.isThrown)
```

File: src/main/java/thePackmaster/cards/rimworldpack/EatWithoutTable.java [hunk 8]
```diff
@@ -34,14 +36,17 @@ public void use(AbstractPlayer p, AbstractMonster m) {
...
+                        drinkable.add(pot);
                 }
+
+                if(drinkable.size() > 0)
+                    potion = drinkable.get(AbstractDungeon.cardRandomRng.random(drinkable.size()-1));
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rimworldpack/EatWithoutTable.java:48]: Make copy of potion

**Review by erasels — CHANGES_REQUESTED** [review]

**Review by erasels — APPROVED** [review]

**Pull request merged**